src/Microsoft.VisualStudio.Services.Agent/ProcessInvoker.cs

```diff
@@ -31,79 +35,58 @@ public interface IProcessInvoker : IDisposable, IAgentService
-                        //TODO: find out what is Process.Kill doing with child processes on OSX and Linux
-                        _proc.Kill();
-                    }
-                    catch (InvalidOperationException ex)
-                    {
-                        //process has exited, before we got a chance to kill it
-                        Trace.Error(ex);
-                    }
-                }
-            }
-            // Wait for process to exit without hard timeout, which will 
-            // ensure that we've read everything from the stdout and stderr.
-            _proc.WaitForExit();
-
-            Trace.Info(
-                "Finished process with file name '{0}', arguments '{1}', exit code {2}, and elapsed time {3}.",
-                _proc.StartInfo.FileName,
-                _proc.StartInfo.Arguments,
-                _proc.ExitCode,
-                _stopWatch.Elapsed);
-            return _proc.ExitCode;
+            return ExecuteAsync(
+                workingDirectory: workingDirectory,
+                fileName: fileName,
+                arguments: arguments,
+                environment: environment,
+                requireExitCodeZero: false,
+                cancellationToken: cancellationToken);
         }
 
-        private void Start(string workingDirectory, string fileName, string arguments, IDictionary<string, string> environment)
+        public async Task<int> ExecuteAsync(
+            string workingDirectory,
+            string fileName,
+            string arguments,
+            IDictionary<string, string> environment,
+            bool requireExitCodeZero,
+            CancellationToken cancellationToken)
         {
             ArgUtil.Null(_proc, nameof(_proc));
             ArgUtil.NotNullOrEmpty(fileName, nameof(fileName));
-            Trace.Info("Starting process with file name '{0}', arguments '{1}', and working directory '{2}'.", fileName, arguments, workingDirectory);
 
-            // Setup the start info.
+            Trace.Info($"Starting process with file name '{fileName}', arguments '{arguments}', and working directory '{workingDirectory}'.");
             _proc = new Process();
             _proc.StartInfo.FileName = fileName;
             _proc.StartInfo.Arguments = arguments;
             _proc.StartInfo.WorkingDirectory = workingDirectory;
             _proc.StartInfo.UseShellExecute = false;
-            _proc.StartInfo.RedirectStandardInput = false;
-            _proc.StartInfo.RedirectStandardError = ErrorDataReceived != null;
-            _proc.StartInfo.RedirectStandardOutput = OutputDataReceived != null;
             _proc.StartInfo.CreateNoWindow = true;
+            _proc.StartInfo.LoadUserProfile = false;
+            _proc.StartInfo.RedirectStandardInput = true;
+            _proc.StartInfo.RedirectStandardError = true;
+            _proc.StartInfo.RedirectStandardOutput = true;
+            _proc.StartInfo.StandardErrorEncoding = Encoding.UTF8;
```

Comment:
What I mean is, you shouldn't set it. You should let it default to whatever it defaults to.